test/Streamly/Test/Data/Unbox.hs

```diff
@@ -157,8 +203,61 @@ checkSizeOf _ size = sizeOf (Proxy :: Proxy a) `shouldBe` size
 -- Tests
 --------------------------------------------------------------------------------
 
+testCustomDatatype1TH :: Spec
+testCustomDatatype1TH = do
+    it "CustomDataType1" $ testSerialization CustomDataType1
+
+    CHECK_SIZE(CustomDataType1, 1)
+
+testCustomDatatype2TH :: Spec
+testCustomDatatype2TH = do
+    it "CustomDataType2 1" $ testSerialization CDT2Constructor1
+    it "CustomDataType2 2" $ testSerialization CDT2Constructor2
+    it "CustomDataType2 3" $ testSerialization CDT2Constructor3
+
+    CHECK_SIZE(CustomDataType2, 1)
+
+testCustomDatatype3TH :: Spec
+testCustomDatatype3TH = do
+    it "CustomDataType3 1"
+           $ testSerialization
+                 (CDT3Constructor1 3 :: CustomDataType3 Int Bool Double)
+    it "CustomDataType3 2"
+           $ testSerialization
+                 (CDT3Constructor2 3 False :: CustomDataType3 Int Bool Double)
+    it "CustomDataType3 3"
+           $ testSerialization (CDT3Constructor3 (3 :: Int) False (5 :: Double))
+
+    checkSizeOfNew
+        "CustomDataType3"
+        (Proxy :: Proxy (CustomDataType3 Int Bool Double))
+        (1 + sizeOf (Proxy :: Proxy Int)
+           + sizeOf (Proxy :: Proxy Bool)
+           + sizeOf (Proxy :: Proxy Double))
+
+testCustomDatatype4TH :: Spec
+testCustomDatatype4TH = do
+    it "CustomDataType4 1"
+           $ testSerialization (CDT4Constructor1 :: CustomDataType4 a Int)
+    it "CustomDataType4 2"
+           $ testSerialization
+                 (CDT4Constructor2 True :: CustomDataType4 a Int)
+    it "CustomDataType4 3"
+           $ testSerialization (CDT4Constructor3 False (5 :: Int))
+
+    checkSizeOfNew
+        "CustomDataType4"
+        (Proxy :: Proxy (CustomDataType4 a Int))
+        (1 + sizeOf (Proxy :: Proxy Bool) + sizeOf (Proxy :: Proxy Int))
+
 testCases :: Spec
 testCases = do
+
+    testCustomDatatype1TH
+    testCustomDatatype2TH
+    testCustomDatatype3TH
+    testCustomDatatype4TH
```

Comment:
Can we have some round tripping tests i.e. serialize/deserialize should provide the same result?

Reply:
It also runs all the tests used for testing the Generic deriving code.